### PR TITLE
LinearCounting recompute size tripping assertion

### PR DESCRIPTION
**Pull request by iverase**
Linear counting algorithm holds two big arrays, one for holding the number of elements per buckets (`sizes`) and another one that holds the values for a given bucket (`hll.runlens`). The issue here is that in elastic/elasticsearch#60201 we changed the way BigArrays are resized. So it might happen that the `sizes` array contains more buckets than `hll.runlens` array. When we recompute the size (in an assertion), we might try to read values for the`hll.runlens` array and trip an out of bounds exception. 

This commit adds a safeguard against this case.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlusTests.java [hunk 1]
```diff
@@ -171,4 +171,16 @@ public long addWithoutBreaking(long bytes) {
 
         assertThat(total.get(), equalTo(0L));
     }
+
+    public void testRetrieveCardinality() {
+        final int p = randomIntBetween(MIN_PRECISION, MAX_PRECISION);
+        final HyperLogLogPlusPlus counts = new HyperLogLogPlusPlus(p, BigArrays.NON_RECYCLING_INSTANCE, 1);
+        int bucket = randomInt(100);
+        counts.collect(bucket, -8688952809613614893L);
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlusTests.java:179]: Probably worth a comment about why this particular random number. If it is to trip the problem, can we somehow assert that it tripped with this number? I feel like we could easily end up in a position where the test doesn't do anything if we don't have some package private method to check to see if this thing happened *somehow*.

**Review by nik9000 — COMMENTED** [review]
Fix LGTM but I left a question about the test.

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlusTests.java (hunk 1 above)
Comment (iverase) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlusTests.java:179]: The only condition to trigger it is to add one value to one bucket. I change it to a random long.

**Review by nik9000 — APPROVED** [review]

**Comment by iverase** [comment]
changing it as non-issue as the regression has not been released.

**Pull request merged**